src/SBP_coefficients/SharanBradyLivescu2022.jl

```diff
@@ -0,0 +1,290 @@
+
+"""
+    SharanBradyLivescu2022(alpha_left, alpha_right)
+
+Coefficients of the cut-cell SBP operators given in
+- Sharan, Brady, Livescu (2022)
+  High-order dimensionally-split Cartesian embedded boundary method for non-dissipative schemes.
+  Journal of Computational Physics 464, 111341.
+
+Here, `alpha_left` * Δx is the spacing between the left endpoint and the second node, while
+`alpha_right` * Δx is the spacing between the right endpoint and the last node.
+"""
+struct SharanBradyLivescu2022{T} <: SourceOfCoefficients
+    alpha_left::T
+    alpha_right::T
+end
+
+function Base.show(io::IO, source::SharanBradyLivescu2022)
+    if get(io, :compact, false)
+      summary(io, source)
+    else
+        print(io,
+        "SharanBradyLivescu2022($(source.alpha_left), $(source.alpha_right)) \n",
+        "   Coefficients of the cut-cell SBP operators given in \n",
+        "    Sharan, Brady, Livescu (2022) \n",
+        "    High-order dimensionally-split Cartesian embedded boundary method for non-dissipative schemes. \n",
+        "    Journal of Computational Physics 464, 111341. \n",
+        "Here, $(source.alpha_left) * Δx is the spacing between the left endpoint and the second node, while \n",
+        "$(source.alpha_right) * Δx is the spacing between the right endpoint and the last node. ")
+    end
+end
+
+# A grid with unequal spacing between the the first and second nodes and the
+# second-to-last and last nodes.
+struct CutCellGrid{T,Grid} <: AbstractArray{T,1}
+    xmin::T
+    xmax::T
+    uniform_interior_grid::Grid # the interior nodes
+end
+
+function CutCellGrid(xmin, xmax, alpha_left, alpha_right, N::Int)
+    @argcheck xmin < xmax
+    @argcheck N > 3
+
+    # build the grid on [-1 + α_left * Δx, 1 - α_right * Δx].
+    Δx = (xmax - xmin) / (N - 3) / (1 + (alpha_left + alpha_right) / (N - 3))
+
+    return CutCellGrid(xmin, xmax, range(xmin + alpha_left * Δx, xmax - alpha_right * Δx, length=N-2))
+end
+
+Base.length(grid::CutCellGrid) = length(grid.uniform_interior_grid) + 2
+
+function Base.getindex(grid::CutCellGrid, i::Int)
+    N = length(grid)
+    @boundscheck begin
+      @argcheck i > 0
+      @argcheck i <= N
+    end
+
+    if i == 1
+        return grid.xmin
+    elseif i == N
+        return grid.xmax
+    else
+      @inbounds x = getindex(grid.uniform_interior_grid, i - 1)
+      return x
+    end
+end
+
+Base.size(grid::CutCellGrid) = (length(grid),)
+Base.step(grid::CutCellGrid) = step(grid.uniform_interior_grid)
```

Comment:
This does not appear to be used currently. Could you please specialize

```julia
construct_grid(source::SharanBradyLivescu2022, accuracy_order, xmin, xmax, N)
```

as done for example in

https://github.com/ranocha/SummationByPartsOperators.jl/blob/34aaa1e87ecd8819c6a27f5b188233caac65c842/src/SBP_coefficients/MattssonAlmquistVanDerWeide2018Accurate.jl#L24

?

Reply:
Done! This also explains the extra test failures when using non-uniform grids. I've amended the tests to use non-uniform cut grids.